modules/core/src/main/java/org/mycontroller/standalone/gateway/philipshue/PhilipsHueGatewayPoller.java

```diff
@@ -0,0 +1,267 @@
+                                _logger.debug("Error while updating hue lights: {} - {} - {} ",
+                                        updateState.getStatusCode(),
+                                        updateState.getErrorMsg(), rawMessage);
+                            }
+                        } else
+                            _logger.warn(" Unable to update {} ", rawMessage);
+                    } catch (Exception ex) {
+                        _logger.error("Exception, ", ex);
+                    }
+                } else if (type == MESSAGE_TYPE.C_INTERNAL) {
+                    switch (MESSAGE_TYPE_INTERNAL.fromString(data.get(3))) {
+                        case I_PRESENTATION:
+                            setOnRequestUpdate(true);
+                            break;
+                        default:
+                            _logger.error(" Unknow internal messager for PhilipsHueGateway {} ", data);
+                            break;
+                    }
+                }
+            } else {
+                _logger.error("data array size should be exactly 4, data:{}", data);
+            }
+        } else {
+            _logger.warn("Private key not set for this {}", gateway);
+        }
+
+    }
+
+    private State getHueUpdateState(MESSAGE_TYPE_SET_REQ subType, List<String> data) {
+        State state = null;
+        switch (subType) {
+            case V_LIGHT_LEVEL:
+                double userValue = Double.valueOf(data.get(1)).doubleValue();
+                state = State.builder().bri(PhilipsHueUtils.toBrightness(userValue).intValue()).build();
+                break;
+            case V_STATUS:
+                state = State.builder().on("1".equalsIgnoreCase(data.get(1))).build();
+                break;
+            case V_RGB:
+                String sensorId = data.get(0);
+                try {
+                    ClientResponse<LightState> lightState = philipsHueClient.lights().state(sensorId);
+                    if (lightState != null && lightState.isSuccess()) {
+                        float[] xy = PHUtilities.calculateXY(
+                                Color.parseColor(data.get(1)), lightState.getEntity().getModelid());
+                        state = State.builder().xy(new Float[] { xy[0], xy[1] }).build();
+                    } else if (lightState != null) {
+                        _logger.debug("Error while getting hue lights state: {} - {} - {} ",
+                                lightState.getStatusCode(),
+                                lightState.getErrorMsg(), data);
+                    }
+                } catch (Exception e) {
+                    _logger.error("Error while getting hue lights state ", e);
+                }
+                waitForHueBridge();
+                break;
+
+            default:
+                break;
+        }
+        return state;
+    }
+
+    private void waitForHueBridge() {
```

Comment:
@Fraid Use one wait function, and reuse everywhere, not need `waitOnError` and `waitForHueBridge`.